### ISSUE TITLE
error when `pynvml.nvmlDeviceGetName` returns an invalid character

### ISSUE THREAD

**Issue by inimaz** [mlco2/codecarbon#697]
* CodeCarbon version: 2.7.1
* Python version: 3.10
* Operating System: WSL

### Description

Codecarbon error when `pynvml.nvmlDeviceGetName` returns an invalid character.

### Full logs
```
2024-10-27 17:04:43.292 [info] Python output: Tracker started.

2024-10-27 17:04:43.350 [info] Python process exited with code 0
2024-10-27 17:05:06.891 [error] Python error: [codecarbon INFO @ 17:05:06] [setup] RAM Tracking...

2024-10-27 17:05:06.891 [error] Python error: [codecarbon INFO @ 17:05:06] [setup] GPU Tracking...

2024-10-27 17:05:06.952 [error] Python error: [codecarbon INFO @ 17:05:06] Tracking Nvidia GPU via pynvml

2024-10-27 17:05:06.953 [error] Python error: [codecarbon DEBUG @ 17:05:06] GPU available. Starting setup

2024-10-27 17:05:06.975 [error] Python error: Traceback (most recent call last):
  File "/home/inigo/personal/vscode-extension-codecarbon/v2/codecarbon/out/scripts/tracker.py", line 6, in <module>

2024-10-27 17:05:06.976 [error] Python error:     tracker = EmissionsTracker()
  File "/home/inigo/.local/lib/python3.10/site-packages/codecarbon/emissions_tracker.py", line 310, in __init__
    self.set_CPU_GPU_ram_tracking()
  File "/home/inigo/.local/lib/python3.10/site-packages/codecarbon/emissions_tracker.py", line 377, in set_CPU_GPU_ram_tracking
    gpu_devices = GPU.from_utils(self._gpu_ids)
  File "/home/inigo/.local/lib/python3.10/site-packages/codecarbon/external/hardware.py", line 121, in from_utils
    return cls(gpu_ids=gpu_ids)
  File "<string>", line 4, in __init__
  File "/home/inigo/.local/lib/python3.10/site-packages/codecarbon/external/hardware.py", line 63, in __post_init__
    self.devices = AllGPUDevices()
  File "/home/inigo/.local/lib/python3.10/site-packages/codecarbon/core/gpu.py", line 212, in __init__
    gpu_device = GPUDevice(handle=handle, gpu_index=i)
  File "<string>", line 8, in __init__

2024-10-27 17:05:06.976 [error] Python error:   File "/home/inigo/.local/lib/python3.10/site-packages/codecarbon/core/gpu.py", line 42, in __post_init__

2024-10-27 17:05:06.976 [error] Python error:     self._init_static_details()
  File "/home/inigo/.local/lib/python3.10/site-packages/codecarbon/core/gpu.py", line 77, in _init_static_details
    self._gpu_name = self._get_gpu_name()
  File "/home/inigo/.local/lib/python3.10/site-packages/codecarbon/core/gpu.py", line 127, in _get_gpu_name
    name = pynvml.nvmlDeviceGetName(self.handle)
  File "/home/inigo/.local/lib/python3.10/site-packages/pynvml/nvml.py", line 1744, in wrapper

2024-10-27 17:05:06.978 [error] Python error:     return res.decode()
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf8 in position 0: invalid start byte
```

**Comment by inimaz** [comment]
Fixed by https://github.com/mlco2/codecarbon/commit/d273ffc6b69c4e2719af6d234027d3f42a2b4d90.
It will be included in next release

**Comment by inimaz** [comment]
Solved in 2.7.2
